projboard/views.py

```diff
@@ -13,20 +15,40 @@ def error_404(request, exception):
 
 
 def search(request):
-    search_title = ''
+    search_input = ''
     articles = []
     message = ''
+    search_method = 'title'
+
     if request.method == 'POST':
-        search_title = request.POST['title']
-        if search_title != '':
-            articles = Article.search_by_title(search_title)
-        else:
+        search_input = request.POST['searchInput']
+        search_method = request.POST['searchOptions']
+        # just to identify in test case if search_input is empty
+        if search_input == '':
             message = "please enter a title!"
 
+        if search_method == 'title':
+            articles = Article.search_by_title(search_input)
+            if len(articles) == 0:
+                message = 'Article didn\'t found'
+        elif search_method == 'subject':
+            subject = Subject.get_subject_by_name(search_input)
+            if subject is None:
+                message = 'Subject Not Valid'
+            else:
+                articles = Article.search_by_subject(subject.id)
+        else:
+            try:
+                user = User.get_user_by_nickname(search_input)
```

Comment:
I do not understand the logic here search 

Looking in template
```
   <select name="searchOptions">
        <option selected="selected" value="title">Title</option>
                         <option value="subject">Subject</option>
                         <option value="user">User</option>
   </select>
```

Is there a missing 
elif search_method == 'user:   ?

Reply:
I believe it is...
There is a dropdown from which we choose, so the only available options are: \
title \
subject \
user \

So instead of elif search_method == user, we can be sure the else at the end will lead us to `user` \
However, for readability (and for future option of letting users insert a text and not just select from a dropdown), I would have state the elif search_method == user